source/PlayerInfo.cpp

```diff
@@ -3110,6 +3110,17 @@ void PlayerInfo::RegisterDerivedConditions()
 	flagshipPlanetProvider.SetHasFunction(flagshipPlanetFun);
 	flagshipPlanetProvider.SetGetFunction(flagshipPlanetFun);
 
+	auto &&flagshipPlanetAttributesProvider = conditions.GetProviderPrefixed("flagship planet attribute: ");
+	auto flagshipPlanetAttributesFun = [this](const string &name) -> bool
+	{
+		if(!flagship || !flagship->GetPlanet())
+			return false;
+		string attribute = name.substr(strlen("flagship planet attribute: "));
+		return !!flagship->GetPlanet()->Attributes().count(attribute);
```

Comment:
dont double ! cancel each others?

Reply:
I'm confused because you can just not do anything and return flagship->GetPlanet()->Attributes().count(attribute);

Reply:
Ah c++ does implicit conversion of int to bool so your right I can just return the count call.